src/averageAges.js

```diff
@@ -54,10 +74,36 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let peopleWithMom;
+  const agesYoungMoms = [];
+  const mothers = people.filter((mother, i, arr) =>
+    arr.some(arrPerson => mother.name === arrPerson.mother)
+  );
+
+  (onlyWithSon)
+    ? peopleWithMom = people.filter((person, i, arr) =>
+      person.mother && arr.find(arrPerson => arrPerson.name === person.mother)
+        && person.sex === 'm')
+    : peopleWithMom = people.filter((person, i, arr) =>
+      person.mother && arr.find(arrPerson => arrPerson.name === person.mother)
+    );
+
+  peopleWithMom.map(person => {
+    agesYoungMoms.push(person.born
```

Comment:
map method return new array,  so we can write the result of the map method into a variable